### PR TITLE
Fix license check

### LINKED ISSUE #27: Inspekt fails with `--fix` parameter
Without `fix` option inspektor run in Travis does not fail, but if I run `inspekt checkall --exclude=.git --fix` in Travis, it silently fails. `autopep8` was installed.

### PR DESCRIPTION
**Pull request by lmr**
Fixes #27, reported by @rdkdd.

### CODE REVIEW

File: inspektor/utils/process.py [hunk 1]
```diff
@@ -67,8 +67,8 @@ def run(cmd, verbose=True, ignore_status=False, shell=False):
     duration = time.time() - start
     result = CmdResult(cmd)
     result.exit_status = p.returncode
-    result.stdout = stdout
-    result.stderr = stderr
+    result.stdout = str(stdout)
+    result.stderr = str(stderr)
```
Comment (ldoktor) [inline inspektor/utils/process.py]: Why is this required? Isn't the return of communicate already string?

File: inspektor/lint.py [hunk 2]
```diff
@@ -63,8 +69,9 @@ def get_opts(self):
                 pylint_args.append('--disable=%s' % self.ignored_errors)
             if self.enabled_errors:
                 pylint_args.append('--enable=%s' % self.enabled_errors)
-            pylint_args.append('--reports=no')
-            if sys.version_info[:2] > (2, 6):
+            if self._pylint_has_option('--reports'):
```
Comment (ldoktor) [inline inspektor/lint.py]: Yep, this should be safest. I'd prefer if we check `--reports=<y` to avoid possible clashes with `--reports-whatever` or even just example `--reports=n` which might be listed in some comment, but that is probably just paranoiac...

File: inspektor/lint.py [hunk 3]
```diff
@@ -45,10 +46,15 @@ def __init__(self, args, logger=logging.getLogger('')):
             self.log.info('Pylint enabled : %s' % self.enabled_errors)
         else:
             self.log.info('Verbose mode, no disable/enable, full reports')
+        help_result = process.run('pylint --help')
```
Comment (ldoktor) [inline inspektor/lint.py]: Well this looks like a global thing and not per-object one. How about moving this to the top and evaluate only once per import?

Comment (lmr) [inline inspektor/lint.py]: good idea!

File: inspektor/utils/process.py (hunk 1 above)
Comment (lmr) [inline inspektor/utils/process.py]: In py3 it's bytes :)

File: inspektor/lint.py (hunk 2 above)
Comment (lmr) [inline inspektor/lint.py]: Fair enough

File: inspektor/utils/process.py (hunk 1 above)
Comment (ldoktor) [inline inspektor/utils/process.py]: OK

File: inspektor/lint.py (hunk 3 above)
Comment (lmr) [inline inspektor/lint.py]: Added a954a8a that implements the suggestions you made.

File: inspektor/lint.py [hunk 4]
```diff
@@ -63,8 +70,9 @@ def get_opts(self):
                 pylint_args.append('--disable=%s' % self.ignored_errors)
             if self.enabled_errors:
                 pylint_args.append('--enable=%s' % self.enabled_errors)
-            pylint_args.append('--reports=no')
-            if sys.version_info[:2] > (2, 6):
+            if self._pylint_has_option('--reports='):
+                pylint_args.append('--reports=no')
+            if self._pylint_has_option('--score='):
```
Comment (ldoktor) [inline inspektor/lint.py]: OK, hopefully they won't change that format 😄

File: inspektor/lint.py [hunk 5]
```diff
@@ -19,6 +19,9 @@
 from pylint.lint import Run
 
 from .path import PathChecker
+from .utils import process
+
+_PYLINT_HELP_TEXT = process.run('pylint --help').stdout
```
Comment (ldoktor) [inline inspektor/lint.py]: I just noticed this outputs unnecessary `Running...` message. Could you please silence it? Apart from that it looks and work good (as far as my old pylint without `--score` says)

Comment (lmr) [inline inspektor/lint.py]: ah, sure

File: inspektor/cli/app.py [hunk 6]
```diff
@@ -31,14 +31,7 @@ def __init__(self):
...
             )
 
-    def initialize_app(self, argv):
-        self.LOG.debug('initialize_app')
-
-    def prepare_to_run_command(self, cmd):
-        self.LOG.debug('prepare_to_run_command %s', cmd.__class__.__name__)
-
     def clean_up(self, cmd, result, err):
-        self.LOG.debug('clean_up %s', cmd.__class__.__name__)
```
Comment (ldoktor) [inline inspektor/cli/app.py]: yep, at this point they are useless...

File: inspektor/path.py [hunk 7]
```diff
@@ -181,6 +181,10 @@ def check_attributes(self, *args):
...
             checks_passed &= getattr(self.path, arg)
-        if checks_passed:
-            self.log.debug('%s: %s', self.label, self.path)
         return checks_passed
+
+    def log_status(self, status):
+        if status == 'PASS':
+            self.log.debug('%s: %s %s', self.label, self.path, status)
+        elif status == 'FAIL':
+            self.log.error('%s: %s %s', self.label, self.path, status)
```
Comment (ldoktor) [inline inspektor/path.py]: Else: custom status or ValueError might unveil typos...

File: inspektor/style.py [hunk 8]
```diff
@@ -83,20 +83,24 @@ def check_file(self, path):
...
         if status != 0:
-            checker.log_status(status='FAIL')
             self.failed_paths.append(path)
+            fix_status = ''
             if AUTOPEP8_CAPABLE:
                 if self.args.fix:
                     self.log.info('Trying to fix errors with autopep8')
                     try:
-                        process.run('autopep8 --in-place --max-line-length=%s --ignore %s %s' % (self.args.max_line_length, self.args.disable, path))
+                        process.run('autopep8 --in-place --max-line-length=%s --ignore %s %s' % (self.args.max_line_length, self.ignored_errors, path))
```
Comment (ldoktor) [inline inspektor/style.py]: This looks like a fix not related to this commit...

Comment (lmr) [inline inspektor/style.py]: Yep, I'm a lazy bastard :) Sorry for that one.
